### PR TITLE
initial stab at dynamic properties check

### PR DESCRIPTION
**Pull request by j-roskopf**
I don't have a high degree in confidence that there aren't a ton of edge cases with all of the checks ATM, but going to work on stronger logic a bit later with the checks + tests

### CODE REVIEW

File: compose-guard/gradle-plugin/src/main/kotlin/com/joetr/compose/guard/ComposeChecks.kt [hunk 1]
```diff
@@ -46,8 +95,10 @@ object ComposeChecks {
                 checkedMetrics.getComposablesReport().restartableButNotSkippableComposables -
                     goldenMetrics.getComposablesReport().restartableButNotSkippableComposables.toSet()
             throw GradleException(
-                "New Composables were added that are restartable but not skippable! " +
-                    difference.joinToString(","),
+                "New Composables were added that are restartable but not skippable! \n" +
+                    difference.joinToString(",") + "\n" +
+                    "More info: https://github.com/androidx/androidx/blob/androidx-main/compose/compiler/design/compiler-metrics.md" +
```
Comment (JJSwigut) [inline compose-guard/gradle-plugin/src/main/kotlin/com/joetr/compose/guard/ComposeChecks.kt:100]: I love adding resources for more info! Yay for helpful messages.

**Review by JJSwigut — APPROVED** [review]
lgtm!
